### PR TITLE
Update to phpunit 8

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes -
| Related issues/PRs | https://github.com/sulu/sulu/pull/4724
| License | MIT
| Documentation PR | -

#### What's in this PR?

Update to phpunit 8:

###### setUp

```php
// Before
function setUp()
// After
function setUp(): void
```

###### tearDown

```php
// Before
function tearDown()
// After
function tearDown(): void
```

###### registerCompilerPass

```php
// Before
function registerCompilerPass(ContainerBuilder $container)
// After
function registerCompilerPass(ContainerBuilder $container): void
```

###### getConfiguration

Update the AbstractExtensionConfigurationTestCase with return types and import them from the correct namespace.

```php
// Before
protected function getConfiguration()
//  After
protected function getConfiguration(): ConfigurationInterface
```

```php
// Before
protected function getContainerExtension()
//  After
protected function getContainerExtension(): ExtensionInterface
```

######  assertInternalType

```php
// Before
assertInternalType('array', 
// After
assertIsArray(
```

```php
// Before
assertInternalType('object', 
// After
assertIsObject(
```

```php
// Before
assertInternalType('int', 
assertInternalType('integer', 
// After
assertIsInt(
```

```php
// Before
assertInternalType('string', 
// After
assertIsString(
```

###### readAttribute, assertAttributeEquals, getObjectAttribute

Replace them with a custom `ReadObjectAttributeTrait` which does a reflection to read the attribute.

###### Replace deprecated expectedExceptionand expectedExceptionMessage

```php
// Before
\n(     \* )@expectedExceptionMessage (.*?)(\n)([\s\S^?]+?)(\{)
// After
$3$4$5\n        \$this->expectExceptionMessage(\'$2\');
```

```php
// Before
\n(     \* )@expectedException (.*?)(\n)([\s\S^?]+?)(\{)
// After
$3$4$5\n        \$this->expectException($2::class);
```

###### assertContains deprecated on strings

Need manually be replaced with `assertStringContainsString` where its string.

###### assertNotContains deprecated on strings

Need manually be replaced with `assertStringNotContainsString` where its string.

###### FrameworkBundle Update

Require compatible compatible phpunit 8 framework bundle version: ^3.4.23

#### Why?

To use the newest phpunit version which is compatible with the lowest supported php version currently (7.2).

#### BC Breaks/Deprecations

As it does not effect the SuluTestBundle classes still old version of phpunit can be used.

#### To Do

- [ ] Create a documentation PR
- [ ] ~~Add breaking changes to UPGRADE.md~~

### CODE REVIEW

File: src/Sulu/Component/DocumentManager/Tests/Unit/NodeManagerTest.php [hunk 1]
```diff
@@ -89,11 +89,10 @@ public function testFindByPath()
 
     /**
      * It should throw an exception if the node1 was not found.
-     *
-     * @expectedException \Sulu\Component\DocumentManager\Exception\DocumentNotFoundException
      */
     public function testFindNotFound()
     {
+        $this->expectException(\Sulu\Component\DocumentManager\Exception\DocumentNotFoundException::class);
```
Comment (danrot) [inline src/Sulu/Component/DocumentManager/Tests/Unit/NodeManagerTest.php]: use at top

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php [hunk 2]
```diff
@@ -73,11 +73,10 @@ public function setUp()
 
     /**
      * It should throw an exception if both path name and node_name options are given.
-     *
-     * @expectedException \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
      */
     public function testExceptionNodeNameAndPath()
     {
+        $this->expectException(\Symfony\Component\OptionsResolver\Exception\InvalidOptionsException::class);
```
Comment (danrot) [inline src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: use at top

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php [hunk 3]
```diff
@@ -88,11 +87,10 @@ public function testExceptionNodeNameAndPath()
 
     /**
      * It should throw an exception if both path name and parent_path options are given.
-     *
-     * @expectedException \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
      */
     public function testExceptionParentPathAndPath()
     {
+        $this->expectException(\Symfony\Component\OptionsResolver\Exception\InvalidOptionsException::class);
```
Comment (danrot) [inline src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: use at top

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php [hunk 4]
```diff
@@ -211,11 +209,10 @@ public function testAutoCreateParent()
 
     /**
      * It should throw an exception if node_name is specified but no parent node is available.
-     *
-     * @expectedException \Sulu\Component\DocumentManager\Exception\DocumentManagerException
      */
     public function testNodeNameButNotParentNode()
     {
+        $this->expectException(\Sulu\Component\DocumentManager\Exception\DocumentManagerException::class);
```
Comment (danrot) [inline src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: use at top

File: src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/AudienceTargetingCacheListenerTest.php [hunk 5]
```diff
@@ -32,7 +35,7 @@ public function testHandle()
         $audienceTargetingCacheListener = new AudienceTargetingCacheListener();
         $audienceTargetingCacheListener->preHandle($this->getCacheEvent($httpCache, $request, $response));
 
-        $this->assertAttributeEquals(false, 'hadValidTargetGroupCookie', $audienceTargetingCacheListener);
+        $this->assertFalse(false, $this->readObjectAttribute($audienceTargetingCacheListener, 'hadValidTargetGroupCookie'));
```
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/AudienceTargetingCacheListenerTest.php]: fixed

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php (hunk 2 above)
Comment (danrot) [reply src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: Did you miss that one?

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php (hunk 3 above)
Comment (danrot) [reply src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: And that one

File: src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php (hunk 2 above)
Comment (alexander-schranz) [reply src/Sulu/Component/DocumentManager/Tests/Unit/Subscriber/Behavior/Path/ExplicitSubscriberTest.php]: yeah missed the \Symfony did fixed it

**Pull request merged**